src/lib.rs

```diff
@@ -89,7 +89,7 @@ where
     enum Unreachable {}
 
     try_array_init(
-        // monomorphise into an unfallible version
+        // monomorphise into an unfailible version
```

Comment:
the word should be _infallible_

Reply:
My bad, I double checked, but I should have triple checked.